src/olympia/devhub/tasks.py

```diff
@@ -180,25 +179,26 @@ def wrapper(task, id_or_path, *args, **kwargs):
 
 
 @validation_task
-@use_primary_db
```

Comment:
Is this still needed?

Reply:
Yes: it needs to use the primary db when fetching the `FileUpload`, which is created before in the view. Otherwise, if there is a slight delay between primary and replica and the task gets fired too early, it might not be able to fetch the `FileUpload`.

One could argue that it needs to be moved to `@validation_task` instead, though.

Reply:
I am wondering if we need this on the scanner tasks too.

Reply:
Absolutely. This hasn't been a problem since they weren't called as real tasks, but as soon as they are they'll need it, so might as well include it now.